### PR TITLE
feat: introduce cluster_token

### PR DESCRIPTION
**Pull request by varnastadeus**

### CODE REVIEW

File: castai/resource_castai_eks_cluster.go [hunk 1]
```diff
@@ -139,44 +147,55 @@ func resourceCastaiEksClusterRead(ctx context.Context, data *schema.ResourceData
...
+		data.Set(FieldEKSClusterAccountId, *resp.JSON200.Eks.AccountId)
+		data.Set(FieldEKSClusterRegion, *resp.JSON200.Eks.Region)
+		data.Set(FieldEKSClusterName, *resp.JSON200.Eks.ClusterName)
+		data.Set(FieldEKSClusterInstanceProfileArn, *resp.JSON200.Eks.InstanceProfileArn)
 	}
 
-	agentToken, err := retrieveAgentToken(ctx, client)
-	if err != nil {
-		return diag.FromErr(err)
+	if _, ok := data.GetOk(FieldEKSClusterAgentToken); !ok {
```
Comment (matas-cast) [inline castai/resource_castai_eks_cluster.go:168]: I'm probably missing something, but why still retrieve the Agent Token if it's switched out for Cluster token in the examples? When would we need it?

Comment (varnastadeus) [reply castai/resource_castai_eks_cluster.go:168]: I kept it to not introduce breaking change, we should remove it with next major version release

File: castai/resource_castai_eks_cluster.go [hunk 2]
```diff
@@ -139,44 +147,55 @@ func resourceCastaiEksClusterRead(ctx context.Context, data *schema.ResourceData
...
+		tkn, err := retrieveAgentToken(ctx, client)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		data.Set(FieldEKSClusterAgentToken, tkn)
+	}
+
+	// Create token only if missing.
+	if _, ok := data.GetOk(FieldEKSClusterToken); !ok {
+		tkn, err := createClusterToken(ctx, client, data.Id())
```
Comment (matas-cast) [inline castai/resource_castai_eks_cluster.go:178]: Could there be issues and is it correct, to create a token on read action?

Comment (varnastadeus) [reply castai/resource_castai_eks_cluster.go]: you are right, moved it to create

**Review by matas-cast — APPROVED** [review]

**Pull request merged**